version/flavor.go

```diff
@@ -0,0 +1,3 @@
+package version
+
+const Flavor string = "community"
```

Comment:
:thinking: how do we replace this in Premium? `sed`?

Reply:
if we declare this as variable it could be replaced in the makefile, as we do for the version variable

Reply:
In the premium version, as the `flavor.go` file just has this, it's a matter of adding this file in the premium repo but with a different content.

I'll give it a try with the Makefile, as long as it means we don't need to keep two separate Makefiles I'm also good with this approach

Reply:
we could also use the same `version.go` file imho since it is already pretty anemic. wdyt?

Reply:
Separating it in two files was suggested by @arbulu89 due to the way how we replace this in the premium repo (we basically overwrite the file we get from the community one). By having it separate we isolate how much code we are re-defining.

I'd like to keep it simple and avoid the ldflags change as it would mean that in order to build the premium binary, we'd need to adjust how we call `make` by adding a variable to the spec file. That would then mean having a separate spec file for the premium repo or using a %define in prjconfig in IBS which IMO is less visible and would not get applied when compiling trento-premium outside the repo.

Reply:
@rtorrero I'm starting to think as we will have 2 different packages (`trento` and `trento-premium`) we will already need a 2nd spec file. This will be needed most probably. In this case, we could just have a "forked" spec file, and do the magic there, having the community as default version.

Reply:
And for @stefanotorresi @fabriziosestito @dottorblaster I'm afraid that you will need to start looking at the OBS system sooner or later, to realize that there are some constraints that will need have ugly patches XD

Reply:
> @rtorrero I'm starting to think as we will have 2 different packages (trento and trento-premium) we will already need a 2nd spec file. This will be needed most probably. In this case, we could just have a "forked" spec file, and do the magic there, having the community as default version.

It makes sense, but this would give us already 3 sources of the code: trento/main, trento/premium-fork and trento-premium/master. There will be design discussions in January, I would postpone this PR until then.